File: java/src/main/java/com/meteroid/api/Connect.java
// this file is @generated
package com.meteroid.api;

import com.meteroid.MeteroidHttpClient;
import com.meteroid.exceptions.ApiException;
import com.meteroid.models.ConnectedAccount;
import com.meteroid.models.ConnectedAccountsResponse;
import com.meteroid.models.CreateConnectedAccountRequest;
import com.meteroid.models.CreateOnboardingLinkRequest;
import com.meteroid.models.OnboardingLinkResponse;

import okhttp3.HttpUrl;

import java.io.IOException;

public class Connect {
    private final MeteroidHttpClient client;

    public Connect(MeteroidHttpClient client) {
        this.client = client;
    }

    /** List all connected accounts for this platform. */
    public ConnectedAccountsResponse listConnectedAccounts() throws IOException, ApiException {

        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/connected-accounts");
        return this.client.executeRequest(
                "GET", url.build(), null, null, ConnectedAccountsResponse.class);
    }

    /**
     * Create a new connected account (Express flow). Returns the account and an onboarding link for
     * the user to complete setup.
     */
    public ConnectedAccount createConnectedAccount(
            final CreateConnectedAccountRequest createConnectedAccountRequest)
            throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/connected-accounts");
        return this.client.executeRequest(
                "POST", url.build(), null, createConnectedAccountRequest, ConnectedAccount.class);
    }

    /** Retrieve a connected account by ID. */
    public ConnectedAccount getConnectedAccount(final String id) throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(String.format("/api/v1/connected-accounts/%s", id));
        return this.client.executeRequest("GET", url.build(), null, null, ConnectedAccount.class);
    }

    /** Revoke a connected account. All associated tokens are invalidated. */
    public void disconnectAccount(final String id) throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(String.format("/api/v1/connected-accounts/%s", id));
        this.client.executeRequest("DELETE", url.build(), null, null, null);
    }

    /**
     * Generate a new onboarding link for a connected account. Any existing unused link is
     * invalidated. The link expires after a configured duration.
     */
    public OnboardingLinkResponse createOnboardingLink(
            final String id, final CreateOnboardingLinkRequest createOnboardingLinkRequest)
            throws IOException, ApiException {
        HttpUrl.Builder url =
                this.client
                        .newUrlBuilder()
                        .encodedPath(String.format("/api/v1/connected-accounts/%s/onboarding", id));
        return this.client.executeRequest(
                "POST",
                url.build(),
                null,
                createOnboardingLinkRequest,
                OnboardingLinkResponse.class);
    }
}

File: python/meteroid/api/connect.py
# this file is @generated

from ..models import (
    ConnectedAccount,
    ConnectedAccountsResponse,
    CreateConnectedAccountRequest,
    CreateOnboardingLinkRequest,
    OnboardingLinkResponse,
)
from .common import ApiBaseAsync, ApiBaseSync


class ConnectAsync(ApiBaseAsync):
    """connect API."""

    async def list_connected_accounts(
        self,
    ) -> ConnectedAccountsResponse:
        """List all connected accounts for this platform."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/connected-accounts",
        )
        return ConnectedAccountsResponse.from_dict(response.json())

    async def create_connected_account(
        self,
        create_connected_account_request: CreateConnectedAccountRequest,
    ) -> ConnectedAccount:
        """Create a new connected account (Express flow). Returns the account
        and an onboarding link for the user to complete setup."""
        response = await self._request_asyncio(
            method="post",
            path="/api/v1/connected-accounts",
            json_body=create_connected_account_request.to_dict(),
        )
        return ConnectedAccount.from_dict(response.json())

    async def get_connected_account(
        self,
        id: str,
    ) -> ConnectedAccount:
        """Retrieve a connected account by ID."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/connected-accounts/{id}",
            path_params={
                "id": id,
            },
        )
        return ConnectedAccount.from_dict(response.json())

    async def disconnect_account(
        self,
        id: str,
    ) -> None:
        """Revoke a connected account. All associated tokens are invalidated."""
        await self._request_asyncio(
            method="delete",
            path="/api/v1/connected-accounts/{id}",
            path_params={
                "id": id,
            },
        )

    async def create_onboarding_link(
        self,
        id: str,
        create_onboarding_link_request: CreateOnboardingLinkRequest,
    ) -> OnboardingLinkResponse:
        """Generate a new onboarding link for a connected account. Any existing
        unused link is invalidated. The link expires after a configured duration."""
        response = await self._request_asyncio(
            method="post",
            path="/api/v1/connected-accounts/{id}/onboarding",
            path_params={
                "id": id,
            },
            json_body=create_onboarding_link_request.to_dict(),
        )
        return OnboardingLinkResponse.from_dict(response.json())


class Connect(ApiBaseSync):
    """connect API."""

    def list_connected_accounts(
        self,
    ) -> ConnectedAccountsResponse:
        """List all connected accounts for this platform."""
        response = self._request_sync(
            method="get",
            path="/api/v1/connected-accounts",
        )
        return ConnectedAccountsResponse.from_dict(response.json())

    def create_connected_account(
        self,
        create_connected_account_request: CreateConnectedAccountRequest,
    ) -> ConnectedAccount:
        """Create a new connected account (Express flow). Returns the account
        and an onboarding link for the user to complete setup."""
        response = self._request_sync(
            method="post",
            path="/api/v1/connected-accounts",
            json_body=create_connected_account_request.to_dict(),
        )
        return ConnectedAccount.from_dict(response.json())

    def get_connected_account(
        self,
        id: str,
    ) -> ConnectedAccount:
        """Retrieve a connected account by ID."""
        response = self._request_sync(
            method="get",
            path="/api/v1/connected-accounts/{id}",
            path_params={
                "id": id,
            },
        )
        return ConnectedAccount.from_dict(response.json())

    def disconnect_account(
        self,
        id: str,
    ) -> None:
        """Revoke a connected account. All associated tokens are invalidated."""
        self._request_sync(
            method="delete",
            path="/api/v1/connected-accounts/{id}",
            path_params={
                "id": id,
            },
        )

    def create_onboarding_link(
        self,
        id: str,
        create_onboarding_link_request: CreateOnboardingLinkRequest,
    ) -> OnboardingLinkResponse:
        """Generate a new onboarding link for a connected account. Any existing
        unused link is invalidated. The link expires after a configured duration."""
        response = self._request_sync(
            method="post",
            path="/api/v1/connected-accounts/{id}/onboarding",
            path_params={
                "id": id,
            },
            json_body=create_onboarding_link_request.to_dict(),
        )
        return OnboardingLinkResponse.from_dict(response.json())
